code/chat.js

```diff
@@ -888,4 +888,30 @@ function postMsg() {
   );
 
   $("#chatinput input").val("");
-}
\ No newline at end of file
+}
+
+window.chat = {
+  setup,
+  show,
+  backgroundChannelData,
+  toggle,
+  getChatPortalName,
+  get _requestAlertsRunning() {
```

Comment:
@johnd0e could these getters work properly?

Reply:
Why not?
(I see massive refactoring, so all this needs to be tested)